client/src/store/historyStore/historyStore.js

```diff
@@ -95,50 +93,43 @@ const actions = {
         const deletedHistory = await deleteHistoryById(history.id, purge);
         commit("deleteHistory", deletedHistory);
         if (getters.firstHistoryId) {
-            return dispatch("setCurrentHistoryId", getters.firstHistoryId);
+            return dispatch("setCurrentHistory", getters.firstHistoryId);
         } else {
             return dispatch("createNewHistory");
         }
     },
-    loadCurrentHistory({ dispatch }) {
-        getCurrentHistoryFromServer().then((history) => dispatch("selectHistory", history));
+    async loadCurrentHistory({ dispatch }) {
+        return getCurrentHistoryFromServer().then((history) => dispatch("selectHistory", history));
     },
     loadHistories({ commit }) {
-        if (!promises.load) {
+        if (!isLoadingHistories) {
             commit("setHistoriesLoading", true);
-            promises.load = getHistoryList()
+            isLoadingHistories = getHistoryList()
                 .then((list) => {
                     commit("setHistories", list);
                 })
                 .catch((err) => {
                     console.warn("loadHistories error", err);
                 })
                 .finally(() => {
-                    promises.load = null;
+                    isLoadingHistories = null;
```

Comment:
This should reset to `false`, not `null`, right?

```suggestion
                    isLoadingHistories = false;
```